course-design.md

```diff
@@ -0,0 +1,214 @@
+  - Online books: [R Packages](https://r-pkgs.org/) 
+  (we differentiate from this amazing book by structuring everything around 
+  this being a course, with exercises, assessments, (group?) projects, 
+  instructions/comments/notes for instructors, etc)
+    - We use the R Packages book as a reference
+    
+## Chapter outline
+
+Course chapters and content need to be strongly aligned with learning outcome/goal.
+Here is a draft outline to guide development.
+
+- Overview of material
+    - How to use, why to use, general details and overall description
+        - Reference "For Instructors" section for instructors to check out
+
+- Syllabus
+    - Overview of course
+    - Why to learn it
+    - Intended learner (personas, assumptions and expectations etc)
+    - What makes this course different
+        - Go over the "Whole Game" chapter of the R Packages book (https://r-pkgs.org/whole-game.html)
+        to give an idea of what it all looks like in the end of this course
+
+- Folder and file structure (RStudio R Projects)
+    - File paths, very basics of the shell (in RStudio)
+    - Making use of the fs package for filesystem management
+    
+- Setting up an R package
+    - Describing what a package is and when or why to make one
+    - Using devtools, usethis for development
+        - e.g. `use_devtools()` and `use_usethis()`.
+    - Developing functions
+        - Basic explanation of what a function is and its components
+        - Actual process of taking code and converting it into a function will
+        be done in next chapter
+    - Making and using datasets
+
+- Function development in a package environment
+    - Process of creating non-function code, checking that it works, then
+      converting it into a function
+        - There are several workflows for this (create in vignette Rmd, make a dev/creating.R
+        script as a development location, developing in the examples Roxygen
+        section, etc). Which to use?
+    - Building functions up slowly, making small targeted functions that build up
+    into a bigger more complex functions
+    - Process control (if-else, stop, return, switch)
+    - Dependency management
+    - Function documentation (with roxygen2), part 1
+        - To use `@examples` to help with creating function
+
+- (Idea) Additional features to make use of
+    - RStudio addins, snippets, rmarkdown templates, templates
+    - rstudioapi? (as a component of the package, maybe to insert text in rstudio, etc)
+    - Things that a package can do aside from functions
+    - Idea: Shiny app here?
+
+- Version control: Using Git and GitHub as a sole user (part 1)
+    - Using Git in RStudio (standard add-commit-history)
+    - Using Git in the Terminal of RStudio (moving in history with checkout,
+    creating branches, adding and updating remotes)
+    - Setting up GitHub for R package (make use of usethis), pushing and pulling
+        - E.g. with pr_* functions from usethis
```

Comment:
```suggestion
        - E.g. with pr_* functions from usethis
    - Emphasize exercises to practice, not showing output of git in code chunks (they are a pain to edit afterwards)
```

Otherwise, make it so Git output is given as output chunk, not from manual copy and paste.